sox/sox.go

```
@@ -76,19 +102,56 @@ func Render(samples []float32, sampleRate float64, opt Options) (*image.Paletted
 		return err
 	}
-	if err := png.Encode(f, img); err != nil {
-		f.Close()
-		os.Remove(path) // don't leave a partial/corrupt file behind
+	tmp := f.Name()
+	renamed := false
+	defer func() {
+		// Keyed on renamed, not on err: a panic in the encoder would leave err
+		// nil and strand the temporary file otherwise.
+		if !renamed {
+			f.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `f.Close` is not checked

(errcheck)